src/index/src/inverted_index/search/fst_apply/intersection_apply.rs

```diff
@@ -32,7 +35,41 @@ pub struct IntersectionFstApplier {
     ranges: Vec<Range>,
 
     /// A list of `Dfa` compiled from regular expression patterns.
-    dfas: Vec<DFA<Vec<u32>>>,
+    dfas: Vec<DfaFstAutomaton>,
+}
+
+struct DfaFstAutomaton(DFA<Vec<u32>>);
```

Comment:
@zhongzc I'm unsure if wrapping the DFA to make it compiled, and upgrade the version of regex-automata, then we can gain the performance improvement. I just make it work. You may help in reviewing the performance part.